src/middleware/queries.ts

```diff
@@ -47,21 +85,109 @@ export function proposalVotes(
       }
     }
   `;
+  const queryV2 = gql`
+    query ProposalVotesQuery(
+      $pipId: String!
+      $vote: Boolean!
+      $count: Int
+      $skip: Int
+      $orderBy: [ProposalVotesOrderBy!]!
+    ) {
+      proposalVotes(
+        filter: { proposalId: { equalTo: $pipId }, vote: { equalTo: $vote } }
+        first: $count
+        offset: $skip
+        orderBy: $orderBy
+      ) {
+        nodes {
+          blockId
+          eventIdx
+          account
+          vote
+          weight
+        }
+      }
+    }
+  `;
 
   return {
-    query,
-    variables,
+    v1: {
+      query,
+      variables,
+    },
+    v2: {
+      request: {
+        query: queryV2,
+        variables: {
+          ...variables,
+          pipId: variables.pipId.toString(),
+          orderBy: [mapProposalVotesOrder(variables.orderBy)],
+        },
+      },
+      mapper: ({ proposalVotes: results }) => {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const { nodes } = results!;
```

Comment:
null assertion doesn't seem right. 

GraphQL has not null in the schema, and I would have thought the types would be alright if we were returning an empty array. If it can be null, `.map` below will cause a run time error.

Reply:
It seems like graphql should always return with having an empty nodes array e.g. `proposalVotes: { nodes: [] }`

I think the no-non-null is OK here, but ideally our types would know that.